jcl/src/openj9.dataaccess/share/classes/com/ibm/dataaccess/ExternalDecimal.java

```diff
@@ -0,0 +1,121 @@
+/*[INCLUDE-IF DAA]*/
+/*******************************************************************************
+ * Copyright IBM Corp. and others 2023
+ *
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/
+ * or the Apache License, Version 2.0 which accompanies this distribution and
+ * is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * This Source Code may also be made available under the following
+ * Secondary Licenses when the conditions for such availability set
+ * forth in the Eclipse Public License, v. 2.0 are satisfied: GNU
+ * General Public License, version 2 with the GNU Classpath
+ * Exception [1] and GNU General Public License, version 2 with the
+ * OpenJDK Assembly Exception [2].
+ *
+ * [1] https://www.gnu.org/software/classpath/license.html
+ * [2] https://openjdk.org/legal/assembly-exception.html
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0-only WITH Classpath-exception-2.0 OR GPL-2.0-only WITH OpenJDK-assembly-exception-1.0
+ *******************************************************************************/
+package com.ibm.dataaccess;
+
+public class ExternalDecimal {
+    /**
+     * Private constructor, class contains only static methods.
+     */
+    private ExternalDecimal() {
+        super();
+    }
+
+    /**
+     * Checks the validity of a External Decimal, returns an integer indicating the status of the External Decimal.
+     *
+     * @param byteArray
+     *            the source array
+     * @param offset
+     *            starting offset of the External Decimal
+     * @param precision
+     *            precision of the External Decimal
+     * @param decimalType
+     *            constant indicating the type of the decimal
+     *
+     * @return the condition code: 0 if all digit codes and the sign valid, 1 if the sign is invalid,
+     *         2 if at least one digit code is invalid, and 3 if sign and at least one digit code is invalid
+     *
+     * @throws NullPointerException
+     *             if <code>byteArray</code> is null
+     * @throws ArrayIndexOutOfBoundsException
+     *             if an invalid array access occurs
+     * @throws IllegalArgumentException
+     *             if the precision is less than 1
+     */
+    public static int checkExternalDecimal(byte[] byteArray, int offset,
+        int precision, int decimalType) {
+
+        if (precision < 1)
+            throw new IllegalArgumentException("Illegal Precision.");
+
+        if ((offset + (CommonData.getExternalByteCounts(precision, decimalType)) > byteArray.length) || (offset < 0))
```

Comment:
In which case we can have offset less than zero ? Asking because, if caller can set the offset to negative and first check passes, we would not do second check. Just a question.

Reply:
Ah right, that should really be 2 separate checks. I have updated it in my latest commit.